src/components/application_manager/test/commands/mobile/reset_global_properties_test.cc

```diff
@@ -0,0 +1,461 @@
+  key_board_properties[am::strings::language] =
+      static_cast<int32_t>(hmi_apis::Common_Language::EN_US);
+  key_board_properties[am::hmi_request::keyboard_layout] =
+      static_cast<int32_t>(hmi_apis::Common_KeyboardLayout::QWERTY);
+  key_board_properties[am::hmi_request::auto_complete_text] = "";
+  ui_msg_params[am::hmi_request::keyboard_properties] = key_board_properties;
+
+  EXPECT_CALL(app_mngr_,
+              ManageHMICommand(HMIResultCodeIs(
+                  hmi_apis::FunctionID::UI_SetGlobalProperties)))
+      .WillOnce(Return(true));
+
+  EXPECT_CALL(app_mngr_,
+              ManageHMICommand(HMIResultCodeIs(
+                  hmi_apis::FunctionID::TTS_SetGlobalProperties)))
+      .WillOnce(Return(true));
+
+  command_->Run();
+}
+
+TEST_F(ResetGlobalPropertiesRequestTest, OnEvent_InvalidEventId_UNSUCCESS) {
+  Event event(hmi_apis::FunctionID::INVALID_ENUM);
+
+  EXPECT_CALL(app_mngr_, ManageMobileCommand(_, _)).Times(0);
+  command_->on_event(event);
+}
+
+TEST_F(ResetGlobalPropertiesRequestTest,
+       OnEvent_UI_SetGlobalProperties_SUCCESS) {
+  Event event(hmi_apis::FunctionID::UI_SetGlobalProperties);
+  (*msg_)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::eType::SUCCESS;
+
+  (*msg_)[am::strings::msg_params][am::strings::properties][0] =
+      mobile_apis::GlobalProperty::VRHELPTITLE;
+
+  EXPECT_CALL(*mock_app_, reset_vr_help_title());
+  EXPECT_CALL(*mock_app_, reset_vr_help());
+
+  EXPECT_CALL(*mock_app_, set_reset_global_properties_active(true));
+
+  smart_objects::SmartObjectSPtr vr_help =
+      ::utils::MakeShared<smart_objects::SmartObject>(
+          smart_objects::SmartType_Map);
+  EXPECT_CALL(*mock_message_helper_, CreateAppVrHelp(_))
+      .WillOnce(Return(vr_help));
+
+  command_->Run();
+
+  event.set_smart_object(*msg_);
+
+  EXPECT_CALL(app_mngr_,
+              ManageMobileCommand(
+                  MobileResultCodeIs(mobile_apis::Result::eType::SUCCESS),
+                  am::commands::Command::ORIGIN_SDL));
+  EXPECT_CALL(*mock_app_, UpdateHash());
+
+  command_->on_event(event);
+}
+
+TEST_F(ResetGlobalPropertiesRequestTest,
+       OnEvent_TTS_SetGlobalProperties_SUCCESS) {
+  Event event(hmi_apis::FunctionID::TTS_SetGlobalProperties);
+  (*msg_)[am::strings::params][am::hmi_response::code] =
+      hmi_apis::Common_Result::eType::UNSUPPORTED_RESOURCE;
+
+  (*msg_)[am::strings::msg_params][am::strings::properties][0] =
+      mobile_apis::GlobalProperty::TIMEOUTPROMPT;
+  (*msg_)[am::strings::msg_params][am::strings::properties][1] =
+      mobile_apis::GlobalProperty::MENUNAME;
+
+  std::vector<std::string> time_out_prompt;
+  time_out_prompt.push_back("time_out");
+  EXPECT_CALL(app_mngr_settings_, time_out_promt())
+      .WillOnce(ReturnRef(time_out_prompt));
+
+  EXPECT_CALL(*mock_app_, set_timeout_prompt(_));
+
+  smart_objects::SmartObjectSPtr prompt =
+      utils::MakeShared<smart_objects::SmartObject>();
+  *prompt = "prompt";
```

Comment:
@vlantonov Maybe I should better use the SmartType_Map like in other instances of this promp setup:
const smart_objects::SmartObjectSPtr so_time_out_prompt =
      utils::MakeShared<smart_objects::SmartObject>(
          smart_objects::SmartType_Map);